cmd/istioctl/inject.go

```diff
@@ -137,7 +140,7 @@ func init() {
 	rootCmd.AddCommand(injectCmd)
 
 	injectCmd.PersistentFlags().StringVar(&hub, "hub", "docker.io/istio", "Docker hub")
-	injectCmd.PersistentFlags().StringVar(&tag, "tag", "0.1", "Docker tag")
+	injectCmd.PersistentFlags().StringVar(&tag, "tag", "", "Docker tag, defaults to same as version")
```

Comment:
You can use the default version SHA directly, e.g.

```golang
injectCmd.PersistentFlags().StringVar(&tag, "tag", version.Info.GitRevision, "Docker tag, defaults to same as version")
```

Note that this particular SHA method won't always work if developer creates a new GIT commit, rebuilds istioctl, but doesn't push a new  proxy/init images with corresponding SHA.